packages/ra-ui-materialui/src/list/DatagridRow.js

```diff
@@ -107,7 +126,6 @@ const DatagridRow = ({
                         className={classes.expandIconCell}
                     >
                         <ExpandRowButton
-                            className={classes.expandButton}
```

Comment:
this was causing the expand button to be always shown expanded (recent regression)